### ISSUE TITLE
Integration tests should not depends on e2e tests

### ISSUE THREAD

**Issue by gavinfish** [kubernetes/kubernetes#100456]
Originally tracked at #89185. After fixing "not include test files in import-boss" in #90380, we have a regression here:
`test/integration/auth/bootstraptoken_test.go` is using `k8s.io/kubernetes/test/e2e/lifecycle/bootstrap`.

We need remove dependencies from integration tests to e2e tests again.

/assign
/sig testing

**Comment by neolit123** [comment]
/sig auth
as per the required change

**Comment by BenTheElder** [comment]
> We need remove dependencies from integration tests to e2e tests.

It would be helpful to outline an argument as to _why_ this is the case. I can imagine reasons, but none are mentioned here.

**Comment by gavinfish** [comment]
> > We need remove dependencies from integration tests to e2e tests.
> 
> It would be helpful to outline an argument as to _why_ this is the case. I can imagine reasons, but none are mentioned here.

Description updated.

**Comment by spiffxp** [comment]
/triage accepted
/priority backlog
/milestone v1.21
This seems a simple enough fix to land (and thank you for doing this) but marking low enough priority to make it clear we may back out and wait for v1.22 if we don't have bandwidth for this

**Comment by BenTheElder** [comment]
So while I think we should probably do this anyhow I want to note:

> Originally tracked at #89185. 

This issue lists the reasons:
> It is hard to move test/e2e/framework into staging (#74352)
> Need to run integration tests always even if PRs are just changing e2e test code

- we have no intention to stage e2e (there's been much discussion about this since then)
- we generally don't want anything non-informational (required jobs) running only sometimes anyhow, it's just not reliable

Decoupling these seems like a good idea from a purely code-organization perspective, but the reasons in the previous issue are no longer valid.

**Comment by spiffxp** [comment]
/milestone v1.22
The PR didn't land in time. At this point I view this as nice to have. Even if it's low impact, let's wait until next release.
